File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.Composite/FreeText.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Web
Imports System.Web.UI

<DefaultProperty("Text")>
 <ToolboxData("<{0}:FreeText runat=server text=""Insert your text here""></{0}:FreeText>")>
 Public Class FreeText
	Inherits Control
	<Bindable(True)>
	<Category("Appearance")>
	<DefaultValue("")>
	<Localizable(True)>
	Public Property Text() As String
		Get
			Return TryCast(ViewState("Text"), [String])
		End Get

		Set(ByVal value As String)
			ViewState("Text") = value
		End Set
	End Property

	Protected Overrides Sub Render(ByVal writer As HtmlTextWriter)
		writer.Write(Text)
	End Sub
End Class

File: aspnet-in-practice/06/CS/CustomControls.Composite/FreeText.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Web;
using System.Web.UI;

namespace CustomControls.Composite
{
	[DefaultProperty("Text")]
	[ToolboxData("<{0}:FreeText runat=server text=\"Insert your text here\"></{0}:FreeText>")]
	public class FreeText : Control
	{
		[Bindable(true)]
		[Category("Appearance")]
		[DefaultValue("")]
		[Localizable(true)]
		public string Text
		{
			get
			{
				return ViewState["Text"] as String;
			}

			set
			{
				ViewState["Text"] = value;
			}
		}

		protected override void Render(HtmlTextWriter writer)
		{
			writer.Write(Text);
		}
	}
}
